modules/auth/schema/creation.go

```diff
@@ -0,0 +1,210 @@
+package schema
+
+import (
+	"context"
+	"errors"
+
+	"github.com/spaceuptech/space-cloud/config"
+)
+
+type creationModule struct {
+	dbType, project, ColName, FieldKey, columnType string
+	currentFieldStruct, realFieldStruct            *schemaFieldType
+	schemaModule                                   *Schema
+}
+
+// SchemaCreation creates or alters tables of sql
+func (s *Schema) SchemaCreation(ctx context.Context, dbType, col, project, schema string) error {
+	crudCol := map[string]*config.TableRule{}
+	crudCol[col] = &config.TableRule{
+		Schema: schema,
+	}
+
+	crud := config.Crud{}
+	crud[dbType] = &config.CrudStub{
+		Enabled:     true,
+		Collections: crudCol,
+	}
+
+	parsedSchema, err := s.parser(crud)
+	if err != nil {
+		return err
+	}
+
+	currentSchema, err := s.Inspector(ctx, dbType, project, col)
+	if err != nil {
+		return err
+	}
+	realSchema := parsedSchema[dbType]
+
+	batchedQueries := []string{}
+
+	for realColName, realColValue := range realSchema {
+		// check if table exist in current schema
+		currentColValue := currentSchema[realColName]
```

Comment:
Should you check if the key is present in `currentSchema`?

Reply:
the key here is the table name
what if the current schema doesn't have the table specified by realColName
as in schema creation, we have not written any query for creating a table
so that' why I wanted to make sure that the table exists before altering the columns of that table
& return an error if not present

Reply:
So this is correct?